drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/TitledAttachmentFileWidget.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.handlers;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.event.logical.shared.ValueChangeHandler;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import org.guvnor.common.services.project.model.WorkspaceProject;
+import org.gwtbootstrap3.client.ui.FormLabel;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.kie.workbench.common.screens.examples.client.wizard.widgets.ComboBox;
+import org.kie.workbench.common.screens.explorer.model.FolderItemType;
+import org.kie.workbench.common.screens.library.api.AssetInfo;
+import org.kie.workbench.common.screens.library.api.AssetQueryResult;
+import org.kie.workbench.common.screens.library.api.ProjectAssetsQuery;
+import org.kie.workbench.common.screens.library.client.screens.assets.AssetQueryService;
+import org.kie.workbench.common.screens.library.client.util.LibraryPlaces;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.ext.widgets.common.client.callbacks.DefaultErrorCallback;
+import org.uberfire.util.URIUtil;
+
+public class TitledAttachmentFileWidget extends Composite implements ValueChangeHandler<String> {
```

Comment:
I fear you have to add MVP and remove the use of `Composite `here.

Reply:
@Rikkola  
Will be fixed with https://issues.jboss.org/browse/DROOLS-3420